recipe-server/client/control_new/components/common/ShieldIdenticon.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default class ShieldIdenticon extends React.PureComponent {
+  static propTypes = {
+    seed: PropTypes.string.isRequired,
+    size: PropTypes.number,
+    className: PropTypes.string,
+  }
+
+  static defaultProps = {
+    size: 64,
+    className: null,
+  }
+
+  render() {
+    const { seed, size, className } = this.props;
+
+    return (
+      <img
+        className={className}
+        src={`/api/v2/identicon/v1:${seed}.svg`}
```

Comment:
Currently this has the `v1:` version of the shield endpoint. If we make changes to shields and have 'generations' of them, in the future we will need to determine which version shield to display.

We can avoid this by saving the entire `src` string with the recipe and simply doing something like `<img src={recipe.iconUrl} />` on the front end.  We can do this a few ways, but I think changing what we store requires backend work, so I'm not positive what is the most flexible approach there.

With all that said, I don't think we anticipate changing these shields anytime soon, so maybe this is a non-issue for now.

Reply:
Having thought on it for a few minutes, it makes more sense to me to save the resulting svg URL and then using that to recall the shield icon.

An idea I liked for the 'create recipe' flow was to add a random shield generator, the premise being that if the user wanted to cycle through a few shields before settling on one they could do so. If we wanted to incorporate something like that, we would need to allow/save `seed` values that weren't just recipe ID's. The current setup would only allow for one shield which is tied to the ID.

Reply:
@mythmon @Osmose @rehandalal If y'all have any thoughts on this, I'd be interested to hear 'em!

Reply:
Now that #992 has landed, we now [have a `identicton_seed` attribute on recipes](https://github.com/mozilla/normandy/pull/992/files#diff-4b47ef5bd0860f8070da44d56085d838R110). We should be using that instead of `id`.